service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderControllerTest.java

```diff
@@ -0,0 +1,114 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.enums.EPOType;
+import uk.gov.hmcts.reform.fpl.enums.ccd.fixedlists.InterimEndDateType;
+import uk.gov.hmcts.reform.fpl.model.Address;
+import uk.gov.hmcts.reform.fpl.model.order.generated.InterimEndDate;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.EPOType.PREVENT_REMOVAL;
+import static uk.gov.hmcts.reform.fpl.enums.EPOType.REMOVE_TO_ACCOMMODATION;
+import static uk.gov.hmcts.reform.fpl.enums.ccd.fixedlists.InterimEndDateType.END_OF_PROCEEDINGS;
+import static uk.gov.hmcts.reform.fpl.enums.ccd.fixedlists.InterimEndDateType.NAMED_DATE;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(UploadDocumentsController.class)
+@OverrideAutoConfiguration(enabled = true)
+class ValidateOrderControllerTest extends AbstractControllerTest {
+
+    @Autowired
+    private Time time;
+
+    ValidateOrderControllerTest() {
+        super("validate-order");
+    }
+
+    @Test
+    void shouldReturnErrorsWhenEPOTypeIsPreventRemovalButAddressIsIncomplete() {
```

Comment:
did you mean _shouldReturnErrorsWhenEPOTypeIsPreventRemovalButAddressIsIncomplete_ or _shouldReturnErrorsWhenEPOTypeIsPresentButAddressIsIncomplete_?

Reply:
I believe we mean the former, this was a test from @Jamesnelson117 so he would have to confirm, just says it is new as I renamed the file

Reply:
@Gogetter test name is fine as it is. When EPO type is 'prevent removal' there is validation in place to ensure address is populated.